File: 881-boats-to-save-people/881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        left, right, ans = 0, len(people) - 1, 0
        people.sort()          
        
        while left <= right:
            if people[left] + people[right] <= limit:
                left += 1
            right -= 1
            ans += 1
            
        return ans
            
                
            
        

File: 881-boats-to-save-people/881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        int num_boats = 0,n = people.length;
        int left_ptr = 0,right_ptr = n - 1;
        Arrays.sort(people);
        
        while(left_ptr <= right_ptr){
            int weight = people[left_ptr] + people[right_ptr];
            
            if(weight <= limit){
                    left_ptr++;
                }
                num_boats++;
                right_ptr--;
        }
        return num_boats;
    }
}